src/FastExpressionCompiler.LightExpression/Expression.cs

```diff
@@ -949,6 +971,46 @@ internal GotoExpression(GotoExpressionKind kind, LabelTarget target, Expression
         }
     }
 
+    public class SwitchCase
+    {
+        public readonly ReadOnlyCollection<Expression> TestValues;
+        public readonly Expression Body;
+
+        public System.Linq.Expressions.SwitchCase ToSwitchCase() =>
+            SysExpr.SwitchCase(Body.ToExpression(), TestValues.Select(x => x.ToExpression()));
+
+        public SwitchCase(Expression body, IEnumerable<Expression> testValues)
+        {
+            Body = body;
+            TestValues = new ReadOnlyCollection<Expression>(testValues.ToArray());
+        }
+    }
+
+    public class SwitchExpression : Expression
+    {
+        public override ExpressionType NodeType { get; }
+        public override Type Type { get; }
+
+        public override SysExpr ToExpression() => SysExpr.Switch(SwitchValue.ToExpression(), DefaultBody.ToExpression(),
+            Comparison, Cases.Select(x => x.ToSwitchCase()));
```

Comment:
Could you use `Tools.Map` instead of `Select`? Generally try to avoid using Linq.